tests/functional/tests/security/test_management_fuzzy.py

```diff
@@ -206,3 +208,105 @@ def test_neg_set_acp_param(pyocf_ctx, cm, cls):
+            cache.set_promotion_policy(i)
+
+
+@pytest.mark.parametrize("cm", CacheMode)
+@pytest.mark.parametrize("cls", CacheLineSize)
+@pytest.mark.security
+def test_neg_set_nhit_promotion_policy_param(pyocf_ctx, cm, cls):
+    """
+    Test whether it is possible to set invalid promotion policy param id for nhit promotion policy
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
+    :return:
+    """
+    # Start cache device
+    cache_device = Volume(S.from_MiB(30))
+    cache = Cache.start_on_device(
+        cache_device, cache_mode=cm, cache_line_size=cls, promotion_policy=PromotionPolicy.NHIT
+    )
+
+    # Set invalid promotion policy param id and check if failed
+    for i in generate_random_numbers(c_uint8):
+        if i in [item.value for item in NhitParams]:
+            continue
+        with pytest.raises(OcfError, match="Error setting promotion policy parameter"):
+            cache.set_promotion_policy_param(i, 1)
+
+
+@pytest.mark.parametrize("cm", CacheMode)
+@pytest.mark.parametrize("cls", CacheLineSize)
+@pytest.mark.security
+def test_neg_set_nhit_promotion_policy_param_trigger(pyocf_ctx, cm, cls):
+    """
+    Test whether it is possible to set invalid promotion policy param TRIGGER_THRESHOLD for
+    nhit promotion policy
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
+    :return:
+    """
+    # Start cache device
+    cache_device = Volume(S.from_MiB(30))
+    cache = Cache.start_on_device(
+        cache_device, cache_mode=cm, cache_line_size=cls, promotion_policy=PromotionPolicy.NHIT
+    )
+
+    # Set to invalid promotion policy trigger threshold and check if failed
+    for i in generate_random_numbers(c_uint32):
+        if i in range(0, 100):
+            continue
+        with pytest.raises(OcfError, match="Error setting promotion policy parameter"):
+            cache.set_promotion_policy_param(NhitParams.TRIGGER_THRESHOLD, i)
+
+
+@pytest.mark.parametrize("cm", CacheMode)
+@pytest.mark.parametrize("cls", CacheLineSize)
+@pytest.mark.security
+def test_neg_set_nhit_promotion_policy_param_threshold(pyocf_ctx, cm, cls):
+    """
+    Test whether it is possible to set invalid promotion policy param INSERTION_THRESHOLD for
+    nhit promotion policy
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
+    :return:
+    """
+    # Start cache device
+    cache_device = Volume(S.from_MiB(30))
+    cache = Cache.start_on_device(
+        cache_device, cache_mode=cm, cache_line_size=cls, promotion_policy=PromotionPolicy.NHIT
+    )
+
+    # Set to invalid promotion policy insertion threshold and check if failed
+    for i in generate_random_numbers(c_uint32):
+        if i in range(2, 1000):
```

Comment:
Statistically there is a chance that every single number returned by `generate_random_numbers()` will be in range between 2 and 1000. It's very little chance, but anyway we can do this much better. How about having smart generator which we can configure to include/exclude certain ranges?
BTW I've noticed that function `generate_random_numbers()` doesn't really return a generator, but a fully allocated list of numbers, which is kind of sad, especially for bigger ranges.

Reply:
I agree. Potentially it could also fix the time issue with security tests. Maybe this allocation is the bottleneck when running full security suite? How about we make an issue about this potential problem and address it in another patch?